tests/storage/datavolume.go

```diff
@@ -415,34 +402,12 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 					return err
 				}, 30*time.Second, 1*time.Second).Should(BeNil())
 
-				vmi := tests.NewRandomVMI()
-
-				diskName := "disk0"
-				vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, v1.Disk{
-					Name: diskName,
-					DiskDevice: v1.DiskDevice{
-						Disk: &v1.DiskTarget{
-							Bus: v1.DiskBusVirtio,
-						},
-					},
-				})
-				vmi.Spec.Volumes = append(vmi.Spec.Volumes, v1.Volume{
-					Name: diskName,
-					VolumeSource: v1.VolumeSource{
-						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
-							ClaimName: dv.ObjectMeta.Name,
-						}},
-					},
-				})
-
-				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("512M")
-
 				vm := tests.NewRandomVirtualMachine(vmi, true)
 				_, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)
 				Expect(err).ShouldNot(HaveOccurred())
 
 				Eventually(func() bool {
-					vm, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+					vm, err := virtClient.VirtualMachine(util.NamespaceTestDefault).Get(vm.Name, &metav1.GetOptions{})
```

Comment:
Why is this better ?

Reply:
I see; I think the "correct" thing to do is to "update" the vmi pointer above, instead of doing this. I mean doing `vmi, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)` in line [406](https://github.com/kubevirt/kubevirt/pull/8084/files#diff-aa884692b6a85018e3059a6e7623fe689a1a06f9d003cdf7fd24ba4bd76fa923R406).

Reply:
I disagree because in our library we assume that users will create the vmi in the namespace that they need without considering the vmi.namespace.
This is why we never modify the vmi.namespace within `libvmi`

Reply:
? I don't understand the explanation.

I'm asking why is it better to conceptually "read VMI from magic namespace" versus "read VMI from the namespace where the VMI was created". I think that's wrong; if there's a magic/default namespace, I should be able to keep avoiding having to use it anywhere.

Reply:
fwiw, I'm with @maiqueb. Keeping a fresh `vm` after creating it is a good practice anyway - you *know* that the old value is out of date.

Reply:
> fwiw, I'm with @maiqueb. Keeping a fresh `vm` after creating it is a good practice anyway - you _know_ that the old value is out of date.

Plus, it has all the defaults KubeVirt sets for the resource.